algosdk/future/transaction.py

```diff
@@ -87,20 +88,22 @@ def get_txid(self):
         txid = base64.b32encode(txid).decode()
         return encoding._undo_padding(txid)
 
-    def sign(self, private_key):
+    def sign(self, private_key, skip_rekey_check=False):
```

Comment:
Can we implement this without adding a public parameter? Maybe making the Signed transaction directly instead of pretending to sign it with a random key?